src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Components/ComponentNodeWriter.cs

```diff
@@ -340,6 +333,40 @@ protected void WriteComponentTypeInferenceMethod(CodeRenderingContext context, C
         }
     }
 
+    protected static void WriteComponentAttributeName(CodeRenderingContext context, ComponentAttributeIntermediateNode attribute, bool allowNameof = true)
+    {
+        if (allowNameof && attribute.BoundAttribute?.ContainingType is string containingType)
+        {
+            containingType = attribute.Annotations[ComponentMetadata.Component.ConcreteContainingType] as string ?? containingType;
+
+            // nameof(containingType.PropertyName)
+            context.CodeWriter.Write("nameof(");
+            TypeNameHelper.WriteGloballyQualifiedName(context.CodeWriter, containingType);
+            context.CodeWriter.Write(".");
+
+            var isSynthesized = attribute.Annotations.TryGetValue(ComponentMetadata.Bind.IsSynthesized, out string _);
```

Comment:
Consider checking the value is equal to `bool.TrueString`, seems that's what we generally do.